app/Providers/AppServiceProvider.php

```diff
@@ -35,5 +37,23 @@ function ($app) use ($socialite) {
                 return $socialite->buildProvider(VATSIMUKProvider::class, $config);
             }
         );
+
+        Http::macro('ukcp', function ($endpoint, $method = 'get', $body = []) {
```

Comment:
I'm not a massive fan of us using a macro in this way. It feels odd.

I would look at creating a UKCP service that is a class with a method for each API call. Keep the API logic for UKCP tucked away in there.

I'd prefer to be able to do something in the controller like;

```
(new UKCP())->getAirfields()->search($search);
```

Reply:
done